compose-backstack/src/main/java/com/zachklipp/compose/backstack/BackstackFrame.kt

```diff
@@ -0,0 +1,44 @@
+package com.zachklipp.compose.backstack
+
+import androidx.compose.runtime.Composable
+
+/**
+ * Models a frame in a Backstack, with a unique [key] to identify it,
+ * and a [Content] function to display it.
+ *
+ * Use [toBackstackModel] to associate any [List] of UI models with
+ * with [Composable] code that can display them, suitable for use
+ * with [Backstack].
+ */
+interface BackstackFrame<out K : Any> {
+  val key: K
+  @Composable fun Content()
+}
+
+inline fun <reified M : Any, K : Any> BackstackFrame(
+  model: M,
+  key: K,
+  crossinline content: @Composable (M) -> Unit
```

Comment:
Why do you need to take both the model and a function that takes the model? If you just take the function, the caller can just capture the model, right?